### PR TITLE
docs: improve wait_for_index metrics description

### PR DESCRIPTION
**Pull request by schmichael**
Old description of `{plan,worker}.wait_for_index` described the metric
in terms of waiting for a snapshot which has two problems:

1. "Snapshot" is an overloaded term in Nomad and operators can't be
   expected to know which use we're referring to here.
2. The most important thing about the metric is what we're waiting *on*
   before taking a snapshot: the raft index of the object to be
   processed (plan or eval).

The new description tries to cram all of that context into the tiny
space provided.

See #5791 for details about the `wait_for_index` mechanism in general.

### CODE REVIEW

File: website/content/docs/operations/metrics.mdx [hunk 1]
```diff
@@ -389,7 +389,7 @@ those listed in [Key Metrics](#key-metrics) above.
 | `nomad.nomad.plan.evaluate`                          | Time elapsed to evaluate a plan                                   | Nanoseconds          | Summary | host                         |
 | `nomad.nomad.plan.queue_depth`                       | Count of evals in the plan queue                                  | Integer              | Gauge   | host                         |
 | `nomad.nomad.plan.submit`                            | Time elapsed for `Plan.Submit` RPC call                           | Nanoseconds          | Summary | host                         |
-| `nomad.nomad.plan.wait_for_index`                    | Time elapsed for the planner to obtain a snapshot                 | Nanoseconds          | Summary | host                         |
+| `nomad.nomad.plan.wait_for_index`                    | Time elapsed the planner waits for plan's index                   | Nanoseconds          | Summary | host                         |
```
Comment (tgross) [inline website/content/docs/operations/metrics.mdx:392]: Given...
> The most important thing about the metric is what we're waiting on
> before taking a snapshot: the raft index of the object to be
> processed (plan or eval).

and that this text will wrap nicely when rendered, I'd be a bit more verbose here and say something like:

> Time elapsed that planner waits for the raft index of the plan to be processed


(And similar for the one below)

Comment (schmichael) [reply website/content/docs/operations/metrics.mdx:392]: For the absolutely silliest reason:

image

Should I go ahead and resize the whole table or just let these two rows be different?

Comment (tgross) [reply website/content/docs/operations/metrics.mdx:392]: Oh, if I hit `<tab>` in emacs markdown mode it just magically aligns the whole table for me. 😀  But yeah that means it'll wrap wildly in an editor. That's unfortunate about markdown tables. Probably not worth worrying about.

Comment (schmichael) [reply website/content/docs/operations/metrics.mdx]: I pushed the wording change. The columns are realigned but the whole row isn't bumped out.

**Review by tgross — APPROVED** [review]
LGTM!

**Pull request merged**